### PR TITLE
Sbiswas96/tiledlayer

### PR DESCRIPTION
**Pull request by sbiswas96**
Please review the sample showing how to add an offline tile package in Qt

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/TileCacheLayer.json [hunk 1]
```diff
@@ -4,5 +4,11 @@
         "TileCacheLayer.h",
         "TileCacheLayer.cpp"
     ],
-    "name" : "Tile cache layer"
+    "name" : "Tile cache layer",
+	"dataItems" : [
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/TileCacheLayer.json]: tabbing issue here.

File: ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/README.metadata.json [hunk 2]
```diff
@@ -1,16 +1,22 @@
...
+        "cache",
+		"layers",
+		"offline",
+		"tile"
     ],
     "redirect_from": "",
     "relevant_apis": [
-        "TODO"
+        "ArcGISTiledLayer",
+		"Basemap",
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/README.metadata.json]: tabbing issue here.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Tile cache layer
+Load an offline copy of a tiled map service as a basemap.
+
+![](screenshot.png)
+
+## Use case
+Constructing an 'ArcGISTiledLayer' from a local tile package (.tpk & .tpkx) allows you to use that basemap tiled service when the application is offline. Dividing a raster into tiles allows the map to provide relevant tiles and level of detail to the user when panning and zooming. For example, when working in an environment that has no connectivity, this could provide access to a map for navigating your surroundings.
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/README.md]: I think these single quotes need to be backticks.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/README.metadata.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    "keywords": [
+        "cache",
+		"layers",
+		"offline",
+		"tile"
+    ],
+    "redirect_from": "",
+    "relevant_apis": [
+        "ArcGISTiledLayer",
+		"Basemap",
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/README.metadata.json]: tabs

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/TileCacheLayer.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
+    "source" :  "TileCacheLayer.qml",
+    "otherSourceFiles" :  [],
+    "name" : "Tile cache layer",
+	"dataItems" : [
+	{
+		"itemId" : "72e703cd01654e7796eb1ae75af1cb53",
```
Comment (anmacdonald) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/TileCacheLayer.json]: tabs

File: ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/Info.plist [hunk 6]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleDisplayName</key>
+	<string>TileCacheLayer</string>
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/Info.plist]: The naming rule is that we use the name defined in the readme design. In this case, `ArcGIS tiled layer (tile cache)`. This name change should be applied throughout.

File: ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/TileCacheLayer.cpp [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  #endif
+
+    return dataPath;
+  }
+} // namespace
+
+TileCacheLayer::TileCacheLayer(QObject* parent /* = nullptr */):
+  QObject(parent)
+{
+  QString fileLocation = QString("%1/ArcGIS/Runtime/Data/tpk/SanFrancisco.tpk").arg(defaultDataPath());
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/TileCacheLayer.cpp]: const

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/README.md (hunk 3 above)
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/README.md]: QML doesn't really have constructors so we generally change the wording from the common design. In this case, something like "Creating an ArcGISTiledLayer..." would be more appropriate.

File: ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/TileCacheLayer.qml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.6
+import Esri.ArcGISRuntime 100.8
+import Esri.ArcGISExtras 1.1
+Rectangle {
```
Comment (ldanzinger) [inline ArcGISRuntimeSDKQt_QMLSamples/Layers/TileCacheLayer/TileCacheLayer.qml]: should be newline after the imports

File: ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/TileCacheLayer.cpp [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "TileCacheLayer.h"
+
+#include"ArcGISTiledLayer.h"
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Layers/TileCacheLayer/TileCacheLayer.cpp]:
```suggestion
#include "ArcGISTiledLayer.h"
```

**Comment by sbiswas96** [comment]
added DisplayAnnotation

**Pull request merged**